src/axom/core/memory_management.hpp

```diff
@@ -179,6 +201,34 @@ inline T* realloc( T* pointer, std::size_t n ) noexcept
   return pointer;
 }
 
+inline void copy( void* dst, void* src, std::size_t numbytes ) noexcept
+{
+#ifdef AXOM_USE_UMPIRE
+  umpire::ResourceManager & rm = umpire::ResourceManager::getInstance();
+  umpire::op::MemoryOperationRegistry & op_registry = umpire::op::MemoryOperationRegistry::getInstance();
+
+  // Note auto is used because the types will change when we update umpire.
+  // At that point we should make them concrete.
+  auto dstStrategy = rm.getAllocator( "HOST" ).getAllocationStrategy();
```

Comment:
Every strategy has the same type, since they are pointers to the base class `umpire::strategy::AllocationStrategy`.

Reply:
So in the version of Umpire we're using the return type here I believe is `std::shared_ptr<AllocationStrategy>` but this has changed in develop to `AllocationStrategy *` which is why I used auto here.

Reply:
Ah, gotcha!

Reply:
Done.